lib/Interpreter/ValueExtractionSynthesizer.cpp

```diff
@@ -353,7 +350,9 @@ namespace {
                                              E).get();
         CallArgs.push_back(AddrOfE);
       }
-      else if (desugaredTy->isPointerType()) {
+      else if (desugaredTy->isPointerType()
```

Comment:
What about desugaredTy->isAnyPointerType()? Could you add the objc test case if you have one?